src/main/java/it/gov/pagopa/pu/fileshare/service/send/SendFileFacadeServiceImpl.java

```diff
@@ -0,0 +1,79 @@
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.security.NoSuchAlgorithmException;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.stereotype.Service;
+import org.springframework.web.multipart.MultipartFile;
+
+@Slf4j
+@Service
+public class SendFileFacadeServiceImpl implements SendFileFacadeService {
+
+  private final UserAuthorizationService userAuthorizationService;
+  private final FileService fileService;
+  private final FileStorerService fileStorerService;
+  private final NotificationService notificationService;
+  private final String sendFolder;
+  private final String archivedSubFolder;
+
+
+  public SendFileFacadeServiceImpl(
+    UserAuthorizationService userAuthorizationService, FileService fileService,
+    FileStorerService fileStorerService,
+    NotificationService notificationService,
+    @Value("${folders.send-file-folder}") String sendFolder,
+    @Value("${folders.process-target-sub-folders.archive}") String archivedSubFolder) {
+    this.userAuthorizationService = userAuthorizationService;
+    this.fileService = fileService;
+    this.fileStorerService = fileStorerService;
+    this.notificationService = notificationService;
+    this.sendFolder = sendFolder;
+    this.archivedSubFolder = archivedSubFolder;
+  }
+
+  @Override
+  public StartNotificationResponse uploadSendFile(Long organizationId, String sendNotificationId, String digest,
+    MultipartFile sendFile, UserInfo user, String accessToken) {
+    userAuthorizationService.checkUserAuthorization(organizationId, user, accessToken);
+    fileService.validateFile(sendFile);
+    String fileName = sendNotificationId+"_"+sendFile.getOriginalFilename();
+
+    try {
+      if(!digest.equals(FileUtils.calculateFileHash(sendFile.getInputStream())))
+        throw new InvalidFileException("Invalid digest");
+    } catch (IOException | NoSuchAlgorithmException e) {
+      throw new FileUploadException(e.getMessage());
+    }
+
+    if(!checkIfAlreadyUploadedOrArchived(organizationId, sendFolder, fileName)) {
+      fileStorerService.saveToSharedFolder(organizationId, sendFile, sendFolder, fileName);
+    }
+
+    LoadFileRequest fileRequest = LoadFileRequest.builder().fileName(fileName).digest(digest).path(sendFolder).build();
+    return notificationService.startNotification(sendNotificationId,organizationId, fileRequest, accessToken);
+  }
+
+  private boolean checkIfAlreadyUploadedOrArchived(Long organizationId, String sendFolder, String fileName) {
+    Path filePath = fileStorerService.buildOrganizationBasePath(organizationId)
+      .resolve(sendFolder);
+    String fileNameCiphered = fileName + AESUtils.CIPHER_EXTENSION;
+    return Files.exists(
+      FileStorerService.concatenatePaths(filePath.toString(), fileNameCiphered))
+      || Files.exists(FileStorerService.concatenatePaths(filePath.resolve(archivedSubFolder).toString(), fileNameCiphered));
+  }
```

Comment:
instead of copy/paste this method, can you move if inside fileStorerService itself re-using it?